File: apps/patter-voice-sdk/libraries/python/getpatter/stt/deepgram.py
"""Deepgram streaming STT for Patter pipeline mode."""

from __future__ import annotations

import os
from typing import ClassVar

from getpatter.providers.deepgram_stt import DeepgramSTT as _DeepgramSTT

__all__ = ["STT"]


class STT(_DeepgramSTT):
    """Deepgram streaming STT.

    Example::

        from getpatter.stt import deepgram

        stt = deepgram.STT()                # reads DEEPGRAM_API_KEY
        stt = deepgram.STT(api_key="dg_...", endpointing_ms=80)
    """

    # Stable provider key for cost attribution / metrics. Matches the
    # entry in ``pricing.py`` so handlers can resolve pricing without
    # falling back to fragile ``__name__`` stripping.
    provider_key: ClassVar[str] = "deepgram"

    def __init__(
        self,
        api_key: str | None = None,
        *,
        language: str = "en",
        model: str = "nova-3",
        encoding: str = "linear16",
        sample_rate: int = 16000,
        endpointing_ms: int = 150,
        utterance_end_ms: int | None = 1000,
        smart_format: bool = False,
        interim_results: bool = True,
        vad_events: bool = True,
    ) -> None:
        key = api_key or os.environ.get("DEEPGRAM_API_KEY")
        if not key:
            raise ValueError(
                "Deepgram STT requires an api_key. Pass api_key='dg_...' or "
                "set DEEPGRAM_API_KEY in the environment."
            )
        super().__init__(
            api_key=key,
            language=language,
            model=model,
            encoding=encoding,
            sample_rate=sample_rate,
            endpointing_ms=endpointing_ms,
            utterance_end_ms=utterance_end_ms,
            smart_format=smart_format,
            interim_results=interim_results,
            vad_events=vad_events,
        )

File: apps/patter-voice-sdk/libraries/typescript/src/stt/deepgram.ts
/** Deepgram streaming STT for Patter pipeline mode. */
import {
  DeepgramSTT as _DeepgramSTT,
  type DeepgramSTTOptions as _DeepgramSTTOptions,
} from "../providers/deepgram-stt";

// Re-export the canonical options type from the provider so callers have a
// single source of truth and can import either path (BUG #13 follow-up).
export type DeepgramSTTOptions = _DeepgramSTTOptions & {
  /** API key. Falls back to DEEPGRAM_API_KEY env var when omitted. */
  apiKey?: string;
  language?: string;
};

/**
 * Deepgram streaming STT.
 *
 * @example
 * ```ts
 * import * as deepgram from "getpatter/stt/deepgram";
 * const stt = new deepgram.STT();              // reads DEEPGRAM_API_KEY
 * const stt = new deepgram.STT({ apiKey: "dg_...", endpointingMs: 80 });
 * ```
 */
export class STT extends _DeepgramSTT {
  static readonly providerKey = "deepgram";
  constructor(opts: DeepgramSTTOptions = {}) {
    const key = opts.apiKey ?? process.env.DEEPGRAM_API_KEY;
    if (!key) {
      throw new Error(
        "Deepgram STT requires an apiKey. Pass { apiKey: 'dg_...' } or " +
          "set DEEPGRAM_API_KEY in the environment.",
      );
    }
    super(
      key,
      opts.language ?? "en",
      opts.model ?? "nova-3",
      opts.encoding ?? "linear16",
      opts.sampleRate ?? 16000,
      {
        endpointingMs: opts.endpointingMs ?? 150,
        utteranceEndMs: opts.utteranceEndMs === null ? null : opts.utteranceEndMs ?? 1000,
        smartFormat: opts.smartFormat ?? false,
        interimResults: opts.interimResults ?? true,
        ...(opts.vadEvents !== undefined ? { vadEvents: opts.vadEvents } : {}),
      },
    );
  }
}
